modules/post/multi/recon/local_exploit_suggester.rb

```diff
@@ -61,7 +70,33 @@ def is_module_platform?(mod)
   end
 
   def is_session_type_compat?(mod)
-    mod.session_compatible? session.sid
+    # These all the `reasons` due to which the module we are checking is not compatible with the current session
+    reasons = mod.session_incompatibility_reasons(session.sid)
+
+    # As we can potentially ignore some `reasons` (e.g. accepting arch values which are, on paper, not compatible),
+    # this keeps track of valid reasons why we will not consider the module that we are evaluating to be valid.
+    valid_reasons = []
+    reasons.each do |reason|
+      # If we are validating e.g. Arch, we want to remove modules that did not pass validation.
+      if !@validate_arch && reason.include?('architecture')
+        vprint_status "#{mod.name} included with override of #{reason}"
+        next
+      end
+      if !@validate_platform && reason.include?('platform')
+        vprint_status "#{mod.name} included with override of #{reason}"
+        next
+      end
+      if !@validate_meterpreter_commands && (reason.start_with?('missing') || reason.start_with?('unloadable'))
+        vprint_status "#{mod.name} included with override of #{reason}"
+        next
+      end
+
+      # If we don't pass verification checks, the reason why a module is not compatible is considered to be `valid`
+      valid_reasons << reason
+    end
+
+    @valid_reasons = valid_reasons
+    valid_reasons.empty?
```

Comment:
This method is defined as part of this module so the signature and response type is reasonable to change. Currently this relies on side-effects and the instance variable being safe to replace each time it is called. Add to that the side-effect is buried a couple layers deep as the consumer of the instance variable is not the caller. Consider returning the set of `valid_reasons`.  This might also suggest a change to the methods name to make the logic when reasoning about results more clear, maybe something like `session_suppression_reasons`

```suggestion
    valid_reasons
```

I suspect the consumer being multiple call away is the reason an instance variable in used here and I understand if the work to refactor is declined.  This pattern though of setting state in a method that by name and usage looks to just provide a `boolean` answer can introduce issues with long term maintenance.